File: server/Models/Bike.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace server.Models;

public class Bike
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string SerialNumber { get; set; }
    public BikeStatus BikeStatus { get; set; }
    public decimal Latitude { get; set; }
    public decimal Longitude { get; set; }
    public Guid? CurrentStationId { get; set; }

    [JsonIgnore]
    public Station? CurrentStation { get; set; }
}

File: client/interfaces/bike.ts
export interface Bike {
  id: string;
  serialNumber: string;
  bikeStatus: BikeStatus;
  latitude: number;
  longitude: number;
  currentStationId?: string;
}

export enum BikeStatus {
  Available = 0,
  Occupied = 1,
  Reserved = 2,
  Unavailable = 3,
  OutOfService = 4,
}

export interface CreateBikeRequest {
  serialNumber: string;
  bikeStatus: BikeStatus;
  latitude: number;
  longitude: number;
  currentStationId?: string;
}

export interface UpdateBikeRequest {
  id: string;
  serialNumber?: string;
  bikeStatus?: BikeStatus;
  latitude?: number;
  longitude?: number;
  currentStationId?: string;
}
